### PR TITLE
Workable (Independent Publisher)

### PR DESCRIPTION
**Pull request by DavidBreathe**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x
Flow 2
Flow
validated successfully
 ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by DavidBreathe** [comment]
@microsoft-github-policy-service agree [company="Breathe IT AS"]

**Comment by DavidBreathe** [comment]
@microsoft-github-policy-service agree

File: independent-publisher-connectors/Workable/Readme.md.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+### Workable Connector
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/Readme.md.txt:1]: Hi @DavidBreathe thank you so much for your new submission.
May I ask you please to follow our template for the readme file [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md) and for raw code view [here](https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/Independent%20Publisher/readme.md)
Please fix the file and include all required fields.

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,1692 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Workable",
+    "description": "The Workable API can be used in the following ways:\n\nExtract information about your:\naccount\nhiring members\npipeline stages\njobs \nscheduled events\njob's and/or candidate's activity\n\nManage your candidates by:\nFetching candidates\nExtracting detail candidate info",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:5]: @DavidBreathe please remove the "API" word and the newline "\n" from the description. The description should meet our [requirements](https://learn.microsoft.com/en-gb/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector).

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,1692 @@
...
+                  },
+                  "description": "Candidate",
+                  "title": "Candidate"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Candidates by ID",
+        "operationId": "Candidates_id",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:557]: @DavidBreathe please use Camel case on all operationsIds. for example here should be "CandidatesId". Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)
please fix all operationIds in the swagger file.

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,1692 @@
...
+                  "description": "Keywords",
+                  "title": "Keywords"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Jobs by shortcode",
+        "description": "Get Jobs by shortcode",
+        "operationId": "Jobs_shortcode",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:960]: @DavidBreathe please use Camel case on all operationsIds. Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1692 @@
...
+                  "description": "Activities",
+                  "title": "Activities"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Jobs Activities by shortcode",
+        "description": "Get Jobs Activities by shortcode: Returns the activity stream of the given job.",
+        "operationId": "Job_activities",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:1155]: @DavidBreathe please use Camel case on all operationsIds. Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1692 @@
...
+                  "description": "Candidate",
+                  "title": "Candidate"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Events by ID",
+        "description": "Get Events by ID: Returns date for the identified event.",
+        "operationId": "Events_id",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:1390]: @DavidBreathe please use Camel case on all operationsIds. Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1692 @@
...
+                  "format": "int32",
+                  "description": "id"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Post a Subscription",
+        "description": "Post a Subscription: Subscribe to an event.",
+        "operationId": "Post_subscription",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:1486]: @DavidBreathe please use Camel case on all operationsIds. Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1692 @@
...
+                  "type": "array",
+                  "items": {},
+                  "description": "Custom attributes"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Custom Attributes",
+        "operationId": "Custom_attributes",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:1549]: @DavidBreathe please use Camel case on all operationsIds. Please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,1692 @@
...
+                      "stage": {
+                        "type": "string",
+                        "description": "Stage",
+                        "title": "Stage"
+                      },
+                      "disqualified": {
+                        "type": "boolean",
+                        "description": "Disqualified",
+                        "title": "Disqualified",
+                        "enum": [
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:114]: @DavidBreathe you have defined the enum type as "boolean"  but the values are defined as strings. please change to [true, false].

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,1692 @@
...
+                      "hired_at": {
+                        "type": "string",
+                        "description": "Hired On",
+                        "title": "Hired On"
+                      },
+                      "sourced": {
+                        "type": "boolean",
+                        "description": "Sourced",
+                        "title": "Sourced",
+                        "enum": [
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiDefinition.swagger.json:133]: @DavidBreathe you have defined the enum type as "boolean" but the values are defined as strings. please change to [true, false].

File: independent-publisher-connectors/Workable/apiProperties.json [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+          "description": "Provide your API Key from Workable",
+          "tooltip": "Provide your API Key from Workable",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      },
+        "Workable API endpoint": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Workable/apiProperties.json:17]: @DavidBreathe can you please explain why you have this property? is it possible to use camel case?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by DavidBreathe** [comment]
Hi Amjed,

I have not updated with a new commit.

I updated the “WorkableApiEndpoint” property to Camel Case. I need it to define the subdomain for the API url “<subdomain>.workable.com”.

David

File: independent-publisher-connectors/Workable/Readme.md.txt (hunk 1 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/Readme.md.txt]: Updated with new readme file from template

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 2 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: API word removed

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 4 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: Updated with Camel case on all operationsIds

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 5 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: Updated with Camel case on all operationsIds

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 7 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: Updated with Camel case on all operationsIds

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 8 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: Updated with Camel case on all operationsIds

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 6 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json]: Updated with Camel case on all operationsIds

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 9 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:114]: Added 
"enum": [
                          "true",
                          "false"
                        ]

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 10 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:133]: Added
"enum": [
"true",
"false"
]

File: independent-publisher-connectors/Workable/apiProperties.json (hunk 11 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiProperties.json]: The property is needed for defining the API subdomain for API endpoint <subdomain>.workable.com which is different for all customers / clients using the service.

**Comment by Amjed-Ayoub** [comment]
Hello @DavidBreathe,
I hope you are doing well.
I am following up with you to see how things are going with preparing your connector.
Please let us know if you have any questions.
Thank you.

**Comment by DavidBreathe** [comment]
I think i have made all the updates required. Are you waiting for anything from me? 😊

David

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 9 above)
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:114]: @DavidBreathe You are adding the values as strings here please change to [true, false].

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 10 above)
Comment (Amjed-Ayoub) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:133]: @DavidBreathe You are adding the values as strings here please change to [true, false].

**Review by Amjed-Ayoub — DISMISSED** [review]

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 9 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:114]: updated

File: independent-publisher-connectors/Workable/apiDefinition.swagger.json (hunk 10 above)
Comment (DavidBreathe) [reply independent-publisher-connectors/Workable/apiDefinition.swagger.json:133]: updated

**Comment by DavidBreathe** [comment]
Ok, think I got it right now. Thanks!

David

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @DavidBreathe,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @DavidBreathe,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by DavidBreathe** [comment]
Thanks for the update Amjed. 😊

David

**Review by sriyen-msft — APPROVED** [review]

**Comment by DavidBreathe** [comment]
Hi,

I found an issue with entering the API key, updated the connector with an new pull request.

https://github.com/microsoft/PowerPlatformConnectors/pull/2172

David

**Comment by DavidBreathe** [comment]
Hi @***@***.***>,

Thank you so much for submitting a new update for your connector.

We will review very soon.

Thank you so much for your patience and understanding.

Best regards,
Amjed Ayoub


From: David Kjell ***@***.***>
Sent: Friday, November 25, 2022 3:43 AM
To: microsoft/PowerPlatformConnectors ***@***.***>; microsoft/PowerPlatformConnectors ***@***.***>
Cc: Mention ***@***.***>; Amjed Ayoub (TEKsystems, Inc.) ***@***.***>
Subject: [EXTERNAL] RE: [microsoft/PowerPlatformConnectors] Workable (Independent Publisher) (PR #2086)

Hi,

I found an issue with entering the API key, updated the connector with an new pull request.

https://github.com/microsoft/PowerPlatformConnectors/pull/2172<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2172&data=05%7C01%7Cv-amjedayoub%40microsoft.com%7C407b115d3cee4be9781508daceda30c2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638049735758781910%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=lmxJVHbTcs9b%2F%2F14VlwB4Ys%2F4O5FEUi%2Fq6yeRsR4tmI%3D&reserved=0>

David

**Pull request merged**